### DISCUSSION TITLE
Visual Studio 2022 17.11.0 Internal Error

### DISCUSSION THREAD

**Discussion by rickpapo** [nlohmann/json#4434] [General]
We ran into a compiler internal error while compiling **json.hpp** with the latest update of Visual Studio 2022, build 17.11.0.  Microsoft has now replicated this error and is investigating.  It might help if other people were to upvote the bug report.

Link: [https://developercommunity.visualstudio.com/t/VS2022-17110-GA-Internal-Compiler-Erro/10722445](https://developercommunity.visualstudio.com/t/VS2022-17110-GA-Internal-Compiler-Erro/10722445)

**Comment by nlohmann** [comment]
Thanks for flagging this and linking the Microsoft Developer Community report — internal compiler errors like this are a compiler bug rather than something fixable in json.hpp itself. If anyone else hits this, please upvote the linked report; if a workaround (e.g., a specific optimization flag or MSVC update) is found, it'd be useful to note it here for others searching this discussion.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
